airbyte-cdk/bulk/core/base/src/main/kotlin/io/airbyte/cdk/AirbyteConnectorRunnable.kt

```diff
@@ -15,18 +16,31 @@ private val log = KotlinLogging.logger {}
 class AirbyteConnectorRunnable : Runnable {
     @Value("\${airbyte.connector.metadata.docker-repository}") lateinit var connectorName: String
 
-    @Inject lateinit var operation: Operation
+    @Inject lateinit var operationProvider: Provider<Operation>
 
     @Inject lateinit var outputConsumer: OutputConsumer
 
+    @Inject lateinit var exceptionClassifier: ExceptionClassifier
+
     override fun run() {
-        log.info { "Executing ${operation::class} operation." }
+        var operation: Operation? = null
         try {
+            try {
+                operation = operationProvider.get()!!
+            } catch (e: Throwable) {
+                throw ConfigErrorException("Failed to initialize connector operation", e)
+            }
```

Comment:
This change now causes Micronaut's instantiation of `Operation` to deferred to this try-catch block which handles any exceptions as ConfigErrorExceptions. This is a requirement to pass the CAT tests, there's a case in there that deliberately attempts a read with an invalid configured catalog and expects a TRACE ERROR message.

Reply:
Will this also mark system errors/other crashes as Config errors?

Reply:
Yes in theory. In practice, exceptions thrown at this point will be triggered by the CLI inputs exclusively.